app/controllers/PaymentsController.scala

```diff
@@ -38,9 +41,12 @@ class PaymentsController @Inject()(
 
   def setup(): Action[AnyContent] = (authenticated andThen registered).async { implicit request =>
     val sdilRef = request.subscription.sdilRef
+    val utr = request.subscription.utr
     val res = for {
-      balance <- sdilConnector.balance(sdilRef, true, request.internalId)
-      nextUrl <- paymentsConnector.initJourney(sdilRef, balance).map(_.nextUrl)
+      balance <- sdilConnector.balance(sdilRef, withAssessment = true, request.internalId)
+      optLastReturn <- getOptLastReturn(utr, request.internalId)
+      optReturnAmount <- getOptLastReturnAmount(sdilRef, request.internalId)
+      nextUrl <- paymentsConnector.initJourney(sdilRef, balance, optLastReturn: Option[SdilReturn], optReturnAmount).map(_.nextUrl)
```

Comment:
I would put the types after the value names on the left hand side of the <- rather than adding the type for `optLastReturn` in the call to `initJourney`.

For example,
`optReturnAmount: Option[SdilReturn] <- getOptLastReturnAmount(sdilRef, request.internalId)`

Adding the return type as mentioned below would be sufficient to just remove this though if you like :)

Reply:
it did not work, results in syntax errors.

Reply:
I'd probably just remove it if you've added the return type to the function call for it, but happy enough for you to leave it in either if you want to.